jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/ByteBufferMessageSink.java

```
@@ -50,7 +48,7 @@ public void accept(Frame frame, Callback callback)
     {
         try
         {
-            long size = (accumulator == null ? 0 : accumulator.getLength()) + frame.getPayloadLength();
+            long size = (accumulator == null ? 0 : accumulator.size()) + frame.getPayloadLength();
             long maxSize = getCoreSession().getMaxBinaryMessageSize();
             if (maxSize > 0 && size > maxSize)
```

Not directly related to your changes here, but I think this case should either throw or also call `fail()`, instead of just failing the callback and returning.

---

You're right, the accumulator should be released in that case. Let me fix this.